civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -777,6 +778,45 @@ function get_civihr_absence_statuses($status_id = NULL) {
     return $status_id_value;
 }
 
+/**
+ * This function returns the sickness reason label for the sickness reason
+ * option value passed to it.
+ * The sickness reason option group is cached to avoid hitting the Db on
+ * every call.
+ *
+ * @param int $sickness_reason_value
+ *
+ * @return string
+ */
+function get_civihr_absence_sickness_reason_label($sickness_reason_value) {
+    $sicknessReasons = &drupal_static(__FUNCTION__);
+
+    if (!isset($sicknessReasons)) {
+        if ($cache = cache_get('civihr_absence_sickness_reasons')) {
+            $sicknessReasons = $cache->data;
+        }
+        else {
+            try {
+                // Civi init
+                civicrm_initialize();
+                $sicknessReasons = CRM_Core_OptionGroup::values('hrleaveandabsences_sickness_reason', FALSE, FALSE, FALSE, NULL, 'label');
+
+                watchdog('DB hit absence sickness reasons', 'DB');
+            }
+
+            catch (CiviCRM_API3_Exception $e) {
+                $error = $e->getMessage();
```

Comment:
you are not handling the error so why catching it? also I don't feel it really needs to catch anything here so just getting the results from civicrm API and checking for count >0 would be enough I think. At the end sickness reason is not something vital for the user to see.